ion/parse_text_simple.go

```diff
@@ -155,6 +153,12 @@ func doClobReplacements(str []byte) []byte {
 			case '\'', '"', '\\':
 				ret = append(ret, next)
 				index++
+			case 'x':
+				// Decode the hex sequence
+				index += 2
+				octet, _ := strconv.ParseUint(string(str[index:index+2]), 16, 8)
```

Comment:
Unprotected array index access after increment.  Also lacking handling if what is being escaped is an invalid hex number.

Reply:
It is technically an invariant from the lexer, would it be okay to panic in this case?

Reply:
Actually, I checked and this is fixed in #20, so unless you object, I would defer to that PR.

Reply:
That's fine.  As to the general question - if it should be an invariant of the lexer then a panic with a message that includes contextual details would be warranted.